cmd/gitserver/server/vcs_syncer_npm_packages.go

```diff
@@ -365,109 +375,40 @@ func decompressTgz(tgzReadSeeker namedReadSeeker, destination string) (err error
 	// down further. So we strip the superfluous directory if we detect one.
 	//
 	// https://github.com/sourcegraph/sourcegraph/pull/28057#issuecomment-987890718
-	var superfluousDirName *string = nil
-	commonSuperfluousDirectory := true
-	err = withTgz(tgzReadSeeker, func(reader *tar.Reader) (err error) {
-		for {
-			header, err := reader.Next()
-			if err == io.EOF {
-				return nil
-			}
-			if err != nil {
-				return errors.Wrapf(err, "failed to read tar file %s", tgzReadSeeker.name)
-			}
-			switch header.Typeflag {
-			case tar.TypeReg:
-				components := strings.SplitN(header.Name, string(os.PathSeparator), 2)
-				if len(components) != 2 {
-					commonSuperfluousDirectory = false
-					return nil
-				}
-				outermostDir := components[0]
-				if superfluousDirName == nil {
-					superfluousDirName = &outermostDir
-				} else if *superfluousDirName != outermostDir {
-					commonSuperfluousDirectory = false
-					return nil
-				}
-			default:
-				continue
-			}
-		}
-	})
+
+	files, err := os.ReadDir(dir)
```

Comment:
```suggestion
	dirEntries, err := os.ReadDir(dir)
```

Calling it `files` makes the subsequent `f.IsDir()` a bit odd to read.

Reply:
2595180f074eb02de83bd77e3f3a101016508c10